`enforce-consistent-line-wrapping` leaving line too long

### Description

The line remains longer than the `printWidth`; I expect this to never happen when using the rule.

### Flavor

JSX

### Code Input

```tsx
console.log(
	<div>
		<div>
			<div>
				<div>
					<div>
						<div>
							<div
								className="
									translate-y-0 opacity-100 transition-all duration-200 ease-out starting:translate-y-1 starting:opacity-0
								"
							/>
						</div>
					</div>
				</div>
			</div>
		</div>
	</div>
);
```

### Expected Behavior

```tsx
The classes would be split up into multiple lines, since the line is longer than the `printWidth`.
```

### Actual Behavior

```tsx
No reports from the rule.
```

### Reproduction URL

https://github.com/abrahamguo/repro/tree/class-wrapping

### ESLint Log

```shell
repro/index.jsx
  10:1  error  This line has a length of 140. Maximum allowed is 120  max-len

✖ 1 problem (1 error, 0 warnings)
```

### ESLint Config

```typescript
import eslintPluginBetterTailwindcss from 'eslint-plugin-better-tailwindcss';
import parser from '@typescript-eslint/parser';

export default {
	files: [`index.jsx`],
	languageOptions: { parser },
	plugins: { 'better-tailwindcss': eslintPluginBetterTailwindcss },
	rules: {
		'max-len': [`error`, { code: 120 }],
		'better-tailwindcss/enforce-consistent-line-wrapping': [
			`error`,
			{ printWidth: 120, group: `never`, indent: `tab` }
		]
	}
};
```

### Versions

- ESLint: 10.2.0
- Parser: `@typescript-eslint/parser`@8.58.2
- Plugin: 4.4.1
- Node: 25.7.0

--- a/docs/rules/enforce-consistent-line-wrapping.md
+++ b/docs/rules/enforce-consistent-line-wrapping.md
@@ -9,6 +9,7 @@ Enforce tailwind classes to be broken up into multiple lines. It is possible to 
 ### `printWidth`
 
   The maximum line length. Lines are wrapped appropriately to stay within this limit. The value `0` disables line wrapping by `printWidth`.
+  Tabs count according to [`tabWidth`](#tabwidth) when evaluating this limit.
 
   **Type**: `number`  
   **Default**: `80`
@@ -48,6 +49,16 @@ Enforce tailwind classes to be broken up into multiple lines. It is possible to 
 
   **Type**: `number | "tab"`  
   **Default**: `2`
+
+<br/>
+
+### `tabWidth`
+
+  Determines how many columns a tab character contributes when checking `printWidth`.
+  This option only affects width calculations and does not change emitted indentation characters.
+
+  **Type**: `number`  
+  **Default**: `1`
 
 <br/>
 

--- a/src/rules/enforce-consistent-line-wrapping.test.ts
+++ b/src/rules/enforce-consistent-line-wrapping.test.ts
@@ -841,6 +841,71 @@ describe(enforceConsistentLineWrapping.name, () => {
     );
   });
 
+  it("should use tabWidth when checking printWidth", () => {
+
+    const dirty = "a b c d";
+    const clean = "\n\ta b c\n\td\n";
+
+    lint(
+      enforceConsistentLineWrapping,
+      {
+        invalid: [
+          {
+            jsx: `() => <img class="${dirty}" />`,
+            jsxOutput: `() => <img class="${clean}" />`,
+            svelte: `<img class="${dirty}" />`,
+            svelteOutput: `<img class="${clean}" />`,
+
+            errors: 1,
+            options: [{ classesPerLine: 0, indent: "tab", printWidth: 10, tabWidth: 4 }]
+          }
+        ]
+      }
+    );
+  });
+
+  it("should default tabWidth to 1 when it is not configured", () => {
+    lint(
+      enforceConsistentLineWrapping,
+      {
+        invalid: [
+          {
+            jsx: `() => <img class="a b c d" />`,
+            jsxOutput: `() => <img class="\n\ta b c d\n" />`,
+            svelte: `<img class="a b c d" />`,
+            svelteOutput: `<img class="\n\ta b c d\n" />`,
+
+            errors: 1,
+            options: [{ classesPerLine: 0, indent: "tab", printWidth: 10 }]
+          }
+        ]
+      }
+    );
+  });
+
+  it("should still ignore printWidth when it is set to 0 even with tabWidth", () => {
+
+    const dirty = "a b c d";
+    const clean = "\n\ta b c\n\td\n";
+
+    lint(
+      enforceConsistentLineWrapping,
+      {
+        invalid: [
+          {
+            jsx: `() => <img class="${dirty}" />`,
+            jsxOutput: `() => <img class="${clean}" />`,
+            svelte: `<img class="${dirty}" />`,
+            svelteOutput: `<img class="${clean}" />`,
+
+            errors: 1,
+            options: [{ classesPerLine: 3, indent: "tab", printWidth: 0, tabWidth: 4 }]
+          }
+        ]
+      }
+    );
+  });
+
   it("should warn if `lineBreakStyle` is likely misconfigured", async () => {
     {
 

--- a/src/rules/enforce-consistent-line-wrapping.test.ts
+++ b/src/rules/enforce-consistent-line-wrapping.test.ts
@@ -883,6 +883,25 @@ describe(enforceConsistentLineWrapping.name, () => {
     );
   });
 
+  it("should not apply tabWidth when indentation uses spaces", () => {
+    lint(
+      enforceConsistentLineWrapping,
+      {
+        invalid: [
+          {
+            jsx: `() => <img class="a b c d" />`,
+            jsxOutput: `() => <img class="\n  a b c d\n" />`,
+            svelte: `<img class="a b c d" />`,
+            svelteOutput: `<img class="\n  a b c d\n" />`,
+
+            errors: 1,
+            options: [{ classesPerLine: 0, indent: 2, printWidth: 10, tabWidth: 8 }]
+          }
+        ]
+      }
+    );
+  });
+
   it("should still ignore printWidth when it is set to 0 even with tabWidth", () => {
 
     const dirty = "a b c d";

--- a/src/rules/enforce-consistent-line-wrapping.test.ts
+++ b/src/rules/enforce-consistent-line-wrapping.test.ts
@@ -975,7 +975,6 @@ describe(enforceConsistentLineWrapping.name, () => {
 
   // #52
   it("should wrap expressions even if `group` is set to `never`", () => {
-
     const expression = "${true ? 'b' : 'c'}";
 
     const correct = dedent`
@@ -997,7 +996,6 @@ describe(enforceConsistentLineWrapping.name, () => {
         ]
       }
     );
-
   });
 
   it("should be possible to change group separation by emptyLines", () => {
@@ -1099,7 +1097,34 @@ describe(enforceConsistentLineWrapping.name, () => {
         ]
       }
     );
+  });
 
+  it("should still start on a new line when `group` is set to `never` except if `preferSingleLine` is enabled", () => {
+    lint(
+      enforceConsistentLineWrapping,
+      {
+        valid: [
+          {
+            angular: `<img class="\n  a b hover:c\n" />`,
+            html: `<img class="\n  a b hover:c\n" />`,
+            jsx: `() => <img class="\n  a b hover:c\n" />`,
+            svelte: `<img class="\n  a b hover:c\n" />`,
+            vue: `<template><img class="\n  a b hover:c\n" /></template>`,
+
+            options: [{ group: "never", preferSingleLine: false, printWidth: 100 }]
+          },
+          {
+            angular: `<img class="a b hover:c" />`,
+            html: `<img class="a b hover:c" />`,
+            jsx: `() => <img class="a b hover:c" />`,
+            svelte: `<img class="a b hover:c" />`,
+            vue: `<template><img class="a b hover:c" /></template>`,
+
+            options: [{ group: "never", preferSingleLine: true, printWidth: 100 }]
+          }
+        ]
+      }
+    );
   });
 
   it("should remove duplicate classes in string literals in defined tagged template literals", () => {

--- a/src/rules/enforce-consistent-line-wrapping.ts
+++ b/src/rules/enforce-consistent-line-wrapping.ts
@@ -598,8 +598,8 @@ class Line {
 
     let width = 0;
 
-    for(const character of line){
-      width += character === "\t"
+    for(let i = 0; i < line.length; i++){
+      width += line[i] === "\t"
         ? tabWidth
         : 1;
     }

--- a/src/rules/enforce-consistent-line-wrapping.ts
+++ b/src/rules/enforce-consistent-line-wrapping.ts
@@ -103,6 +103,14 @@ export const enforceConsistentLineWrapping = createRule({
         description("Enable this option if prettier is used in your project.")
       ),
       "strict"
+    ),
+    tabWidth: optional(
+      pipe(
+        number(),
+        minValue(1),
+        description("The visual width of a tab character when evaluating printWidth.")
+      ),
+      1
     )
   }),
 
@@ -239,8 +247,7 @@ function lintLiterals(ctx: Context<typeof enforceConsistentLineWrapping>, litera
 
           const simulatedLine = multilineClasses.line
             .clone()
-            .addClass(className)
-            .toString();
+            .addClass(className);
 
           // wrap after the first sticky class
           if(
@@ -582,7 +589,22 @@ class Line {
   }
 
   public get length() {
-    return this.toString().length;
+    const line = this.toString();
+    const { tabWidth } = this.ctx.options;
+
+    if(tabWidth <= 1 || !line.includes("\t")){
+      return line.length;
+    }
+
+    let width = 0;
+
+    for(const character of line){
+      width += character === "\t"
+        ? tabWidth
+        : 1;
+    }
+
+    return width;
   }
 
   public get classCount() {

